libxrdp/xrdp_orders_rail.c

```diff
@@ -256,6 +262,8 @@ xrdp_orders_send_window_new_update(struct xrdp_orders *self, int window_id,
     {
         /* titleInfo */
         num_chars = g_mbstowcs(0, window_state->title_info, 0);
+        num_chars = MIN(num_chars, sizeof(twchar) * (g_strlen(window_state->title_info) + 1));
+        num_chars = MAX(num_chars, 0);
```

Comment:
That `MAX` is not giving me confidence. How can `num_chars` be negative? Would it happen if `g_mbstowcs` returns -1 or when `sizeof(twchar) * (g_strlen(window_state->title_info) + 1)` overflows? Let's remove that line. If error handling is needed, it should be explicit. The same applies to every other `MAX`. At least please add a comment.